pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/renderer/TextValueProvider.java

```diff
@@ -1,101 +1,224 @@
+        Map<String, Object> answerObject = getRelevantAnswer(moduleMap, filterConfig);
+        Object rawValue = extractValuesFromAnswer(answerObject, filterConfig);
+        if (answerObject != null) {
+            Object optionDetails = answerObject.get(ESObjectConstants.OPTIONDETAILS);
+            if (optionDetails != null && !((List<?>) optionDetails).isEmpty()) {
+                filterConfig.setHasDetails(true);
+            }
+        }
+
+        if (!(rawValue instanceof List)) {
+            return Collections.singletonList(rawValue);
+        }
+        return (List) rawValue;
     }
 
-    protected List<String> getOptionDetails(List<?> optionDetails) {
-        return optionDetails.stream().map(optionDetail ->
-                        new StringBuilder(((Map) optionDetail).get(ESObjectConstants.OPTION).toString())
-                                .append(':')
-                                .append(((Map) optionDetail).get(ESObjectConstants.DETAIL)).toString())
-                .collect(Collectors.toList());
+    protected Map<String, Object> getRelevantAnswer(Map<String, Object> moduleMap, FilterExportConfig filterConfig) {
+        if (moduleMap != null) {
+            List<Map<String, Object>> allAnswers =
+                    (List<Map<String, Object>>) moduleMap.get(ElasticSearchUtil.QUESTIONS_ANSWER);
+            if (allAnswers != null) {
+                return allAnswers.stream()
+                        .filter(ans -> filterConfig.getColumn().getName().equals(ans.get(ESObjectConstants.STABLE_ID)))
+                        .findAny().orElse(null);
+            }
+        }
+        return null;
+    }
+
+    protected Object extractValuesFromAnswer(Map<String, Object> targetAnswer, FilterExportConfig filterConfig) {
+        if (targetAnswer == null) {
+            return null;
+        }
+        return targetAnswer.getOrDefault(ESObjectConstants.ANSWER, targetAnswer.get(filterConfig.getColumn().getName()));
     }
 
     protected Collection<?> mapToCollection(Object o) {
         if (o == null) {
             return Collections.singletonList(StringUtils.EMPTY);
         }
         if (o instanceof Collection) {
-            if (((Collection<?>) o).isEmpty()) {
+            Collection<?> objList = (Collection<?>) o;
+            if (objList.isEmpty()) {
                 return Collections.singletonList(StringUtils.EMPTY);
             }
-            return (Collection<?>) o;
+            List<Object> allValues = new ArrayList<>();
+            // flatten any nested lists
+            for (Object item : objList) {
+                if (item instanceof Collection) {
+                    allValues.addAll(flatten((Collection) item));
+                } else {
+                    allValues.add(item);
+                }
+            }
+            // replace any nulls with empty string
+            return allValues.stream().map(val -> val == null ? StringUtils.EMPTY : val).collect(Collectors.toList());
```

Comment:
having method with name `replaceNullsWithEmptyString` which would wrap this line would be way more cleaner than having comment.

Reply:
This will also benefit from the Objects::nonNull you pointed me to